### ISSUE TITLE
private-dev disables access for U2F

### ISSUE THREAD

**Issue by hlieberman** [netblue30/firejail#1553]
Because private-dev doesn't allow access to raw USB HID devices, it's currently not possible to use a firejailed version of a browser.

I think it makes a lot of sense for private-dev to restrict access to these devices in the general case, but perhaps a way to allow you to white-list additional devices would be a way to work around this issue?  That way, browsers could opt-in to slightly increase their attack surface by just allowing usb HID access (/dev/usb/hiddev*), as opposed to disabling private-dev altogether.

**Comment by SpotComms** [comment]
See #1176 and #1381 
Duplicate of #1446

**Comment by hlieberman** [comment]
@SpotComms, forgive me, but... #1446 doesn't seem to indicate you're able to mix whitelisting with private-dev.

When I try the simple thing of
```
whitelist /dev/usb/hiddev0
private-dev
```

It launches Firefox in a way that it can't write to the screen, and quickly crashes.  I don't think you can whitelist devices inside of private-dev?

**Comment by SpotComms** [comment]
Yes, #1446 was closed but the idea (fine-grained control of /dev) has still not been implemented. `private-dev` doesn't take into account `whitelist`. For now the only option is just commenting `private-dev`.

**Comment by smitsohu** [comment]
> I don't think you can whitelist devices inside of private-dev?

It is actually possible to run
```
whitelist /dev/
whitelist /dev/

**Comment by netblue30** [comment]
I thinks we should do it. What entries in /dev are these devices bringing in?

**Comment by hlieberman** [comment]
/dev/usb/hiddev[0-9]+.

Both FF (nightly) and Chrome will need it.

**Comment by hlieberman** [comment]
For the record, the whitelisting trick doesn't work.  I'm not sure exactly why; I can see the device when I navigate to /dev/usb through the browser, but it doesn't work even on u2f demo pages.  (https://demo.yubico.com/u2f).

**Comment by chiraag-nataraj** [comment]
Solved with 197bcc86c1a1f4c75d6a42a850e0619b3268db1e (read the comments at #1990 if you're curious as to why the Yubikey doesn't work if you remove it and plug it back in while the `firejail`'d program is running).
